### PR TITLE
sidekiq/util has been removed in 6.5.0

### PR DESCRIPTION
**Pull request by bf4**
see https://github.com/mperham/sidekiq/compare/v6.4.2...v6.5.0

Follows https://github.com/mperham/sidekiq/pull/5340

### CODE REVIEW

**Comment by bf4** [comment]
I think this broke our sidekiq when I deployed it. Hold on

```
NoMethodError: undefined method call' for nil:NilClass (Most recent call first) Show 3 non-project frames NoMethodError: undefined method call' for nil:NilClass (Most recent call first)
Hide 4 non-project frames
File /app/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.5.0/lib/sidekiq/processor.rb line 73 in rescue in run
File /app/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.5.0/lib/sidekiq/processor.rb line 67 in run
File /app/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.5.0/lib/sidekiq/component.rb line 8 in watchdog
File /app/vendor/bundle/ruby/3.0.0/gems/sidekiq-6.5.0/lib/sidekiq/component.rb line 17 in block in safe_thread
```

File: lib/sidekiq/worker_killer.rb [hunk 1]
```diff
@@ -1,12 +1,18 @@
 require "get_process_mem"
 require "sidekiq"
-require "sidekiq/util"
```
Comment (bf4) [inline lib/sidekiq/worker_killer.rb:3]: added in  https://github.com/klaxit/sidekiq-worker-killer/pull/14 https://github.com/klaxit/sidekiq-worker-killer/pull/12

xref 
https://gitlab.com/gitlab-org/gitlab-foss/-/blob/48d25238c386a89e8a6af218eeb290936a8f7595/lib/gitlab/sidekiq_middleware/memory_killer.rb

**Comment by mperham** [comment]
Since worker_killer is a server middleware, you should include the new Module as noted here:

https://github.com/mperham/sidekiq/blob/main/docs/middleware.md#updated-api

You can then access the identity with `config[:identity]`. It looks like that's all you need `sidekiq/util` for.

**Comment by bf4** [comment]
@mperham Thanks. Updated the PR and added a test for finding the process from the processset for the 'identity' to ensure it doesn't fail again

File: lib/sidekiq/worker_killer.rb [hunk 2]
```diff
@@ -125,6 +131,10 @@ def sidekiq_process
     end || raise("No sidekiq worker with identity #{identity} found")
   end
 
+  def identity
+    config[:identity] || config["identity"]
+  end unless method_defined?(:identity)
```
Comment (bf4) [inline lib/sidekiq/worker_killer.rb:136]: include Sidekiq::Util  will have defined this. Else we define it and call the interface added by Sidekiq::ServerMiddleware

**Comment by mperham** [comment]
Why not just have a new version that only supports 6.5+? Let people use an old version of swk if they want to run an old version of Sidekiq.

**Comment by bf4** [comment]
re "Why not just have a new version that only supports 6.5+? Let people use an old version of swk if they want to run an old version of Sidekiq."

I'm not a maintainer of this.  👍 on that, though

**Comment by bf4** [comment]
Still getting failures.  Gotta look deeper. maybe something about the heartbeat? maybe something redis-specific which is why it's not showing up in tests?

https://github.com/klaxit/sidekiq-worker-killer/pull/21#issuecomment-1151124349

looks like somehow Processor is getting a nil `@callback`

I'm trying to reproduce it locally and get more context

my local gem changes, fwiw

sidekiq 6.4.2 -> 6.5.0
sidekiq-pro 5.3.1 -> 5.5.0

**Comment by bf4** [comment]
@mperham Thanks for that.  I thought I might have noticed it on local server shutdown but hadn't reproduced it yet

https://github.com/scoutapp/scout_apm_ruby/issues/449
https://github.com/mperham/sidekiq/issues/5375

**Review by slayer — APPROVED** [review]

**Comment by slayer** [comment]
> Why not just have a new version that only supports 6.5+? Let people use an old version of swk if they want to run an old version of Sidekiq.

It's a bad idea to add breaking changes to minor update, isn't ?

**Comment by mperham** [comment]
Sidekiq::Processor is not a public API.

**Comment by slayer** [comment]
> Sidekiq::Processor is not a public API nor is sidekiq/util.

so why so many gems were broken? (rhetorical question)  :)

**Comment by Laykou** [comment]
What needs to be done to merge this and release a new gem version, please?

**Review by slayer — APPROVED** [review]

**Comment by pcomans** [comment]
@slayer @bf4 is this ready to be merged? If not, how can I help?

**Comment by bf4** [comment]
I'm running this fork in prod.  I am not a maintainer can't merge or release

**Comment by themilkman** [comment]
@ccyrille Any chance for a new release?

**Review by ccyrille — APPROVED** [review]
Hi @bf4, thanks a lot for your contribution and sorry for the late reply! LGTM! I will release a new version of the gem today (cc @slayer @themilkman).

Please note that we are not using `sidekiq-worker-killer` internally at Klaxit anymore. Maybe it would be a good to add external maintainers. Is it something you could be interested in ?

**Comment by bf4** [comment]
@ccyrille Sure, I'm up to add some bandwidth in both github and rubygems, if you're up for it.  Is really just in maintenance mode anyhow.

**Pull request merged**